bodhi/server/util.py

```diff
@@ -315,24 +316,54 @@ def sanity_check_repodata(myurl, repo_type):
 
         if repo_type in ('yum', 'source'):
             tests.append((['list', 'available'], 'testrepo'))
+            tests.append((['updateinfo'], '.*'))
         elif repo_type == 'module':
             tests.append((['module', 'list'], '.*'))
 
+        cfg_options = {
+            '--setopt=zchunk': ['0', '1'],
+        }
+
+        cfg_combinations = build_cfg_combinations(cfg_options)
+
         for test in tests:
             dnfargs, expout = test
 
-            # Make sure every DNF test runs in a new temp dir
-            testdir = tempfile.mkdtemp(dir=tmpdir)
-            output = sanity_check_repodata_dnf(testdir, myurl, *dnfargs)
-            if (expout == ".*" and len(output.strip()) != 0) or (expout in output):
-                continue
-            else:
-                raise RepodataException(
-                    f"DNF did not return expected output when running test!"
-                    + f" Test: {dnfargs}, expected: {expout}, output: {output}")
+            for config in cfg_combinations:
+                config = list(config)
+
+                # Make sure every DNF test runs in a new temp dir
+                testdir = tempfile.mkdtemp(dir=tmpdir)
+                output = sanity_check_repodata_dnf(testdir, myurl, config, *dnfargs)
+                if (expout == ".*" and len(output.strip()) != 0) or (expout in output):
+                    continue
+                else:
+                    raise RepodataException(
+                        f"DNF did not return expected output when running test!"
+                        + f" Test: {dnfargs}, config: {config}, expected: {expout},"
+                        + f" output: {output}")
+
+
+def build_cfg_combinations(options):
+    """
+    Builds the Cartesian product of cfg_options
+
+    Args:
+        options (dict): A dict of option -> [values].
+    Returns:
+        list: A list of all config option combinations to test.
+    """
+    raw_options = []
+    for option in options:
+        raw_option = []
+        for value in options[option]:
+            raw_option.append("%s=%s" % (option, value))
+        raw_options.append(raw_option)
+
+    return list(itertools.product(*raw_options))
```

Comment:
Instead of converting the list elements to lists themselves later, here you may want to do something like:
```python
return [list(result) for result in itertools.product(*raw_options)]
```